test/data/fakedio.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:dio/dio.dart';
+
+import 'test_data.dart';
+
+// to be used later
+class FakeDio implements Dio {
+  FakeDio();
+
+  @override
+  Future<Response<T>> get<T>(
```

Comment:
@alexdivadi  thanks for looking into it,, just remember with retrofit it's fetch instead of get.